tests/matrix_free/matrix_vector_rt_face_common.h

```diff
@@ -0,0 +1,301 @@
+  }
+
+  Vector<Number> solution, initial_condition;
+
+  // create vector with random entries
+  mf_data.initialize_dof_vector(solution);
+  mf_data.initialize_dof_vector(initial_condition);
+
+  for (unsigned int i = 0; i < dof.n_dofs(); ++i)
+    {
+      if (constraints.is_constrained(i))
+        continue;
+      initial_condition[i] = random_value<Number>();
+    }
+
+  MatrixFreeTest<dim, fe_degree, fe_degree + 2, Number> mf(mf_data);
+  mf.test_functions(solution, initial_condition);
+
+
+  SparsityPattern        sp;
+  SparseMatrix<double>   system_matrix;
+  DynamicSparsityPattern dsp(dof.n_dofs(), dof.n_dofs());
+  DoFTools::make_sparsity_pattern(dof, dsp);
+  sp.copy_from(dsp);
+  system_matrix.reinit(sp);
+
+  FEFaceValues<dim> fe_val(
+    dof.get_fe(),
+    QGaussLobatto<dim - 1>(mf_data.get_quadrature(0).size()),
+    update_values | update_gradients | update_JxW_values);
+
+
+  const unsigned int dofs_per_cell = fe_val.get_fe().dofs_per_cell;
+  std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
+  FullMatrix<double> local_matrix(dofs_per_cell, dofs_per_cell);
+
+
+  const FEValuesExtractors::Vector velocities(0);
+  // Assemble matrix
+  for (const auto &cell : dof.active_cell_iterators())
+    for (const auto &face : cell->face_iterators())
+      {
+        fe_val.reinit(cell, face);
+        local_matrix = 0;
+
+        for (const auto q : fe_val.quadrature_point_indices())
+          for (unsigned int i = 0; i < dofs_per_cell; ++i)
+            {
+              const Tensor<1, dim> phi_i = fe_val[velocities].value(i, q) * 10.;
+              const Tensor<2, dim> grad_phi_i =
+                fe_val[velocities].gradient(i, q);
+
+              for (unsigned int j = 0; j < dofs_per_cell; ++j)
+                {
+                  const Tensor<1, dim> phi_j = fe_val[velocities].value(j, q);
+                  const Tensor<2, dim> grad_phi_j =
+                    fe_val[velocities].gradient(j, q);
+                  local_matrix(i, j) +=
+                    (phi_j * phi_i + scalar_product(grad_phi_i, grad_phi_j)) *
+                    fe_val.JxW(q);
+                }
+            }
+        cell->get_dof_indices(local_dof_indices);
+        for (unsigned int i = 0; i < dofs_per_cell; ++i)
+          for (unsigned int j = 0; j < dofs_per_cell; ++j)
+            system_matrix.add(local_dof_indices[i],
+                              local_dof_indices[j],
+                              local_matrix(i, j));
+      }
+
+  Vector<Number> ref(solution.size());
+
+  // Compute reference
+  system_matrix.vmult(ref, initial_condition);
+
+  ref -= solution;
+
+  const double diff_norm = ref.linfty_norm() / solution.linfty_norm();
+  deallog << "Norm of difference: " << diff_norm << std::endl << std::endl;
+}
+
+
+int
+main()
+{
+  initlog();
+
+  {
+    deallog.push("2d");
+    test<2, 2>();
+    test<2, 3>();
+    deallog.pop();
+    deallog.push("3d");
+    test<3, 2>();
+    // test<3, 3>(); Takes way too long
```

Comment:
8 cells and 756 dofs, so not that much. But the debug version takes 400 seconds which I'm guessing is because it is so slow to assemble the matrix. Haven't looked into it though.

Reply:
Probably! But how do I create a mesh with only 2 cells..? I know you can refine different cells but not how I would refine a cell into 2 instead of 8 for `dim=3`

Reply:
https://www.dealii.org/developer/doxygen/deal.II/namespaceGridGenerator.html#ac76417d7404b75cf53c732f456e6e971 with [2,1,1] subdivisions.

Reply:
It still takes 200+ seconds for debug+release with 2 cells, so I'll leave this out for now, but I can mention this in the issue with todos. Maybe one could do something different than comparing with `FEValues`?

Reply:
That is strange. Let's keep the 3D test case for now commented. But maybe you could investigate later what takes so long in the case of `FEFaceValues`?

> Maybe one could do something different than comparing with FEValues?

Alternatively, one can solve a PDE and check the convergence rate.